File: 20.valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        parenMap = {
            "[": "]",
            "{": "}",
            "(": ")",
        }
        for c in s:
            if c in parenMap.keys():
                stack.append(c)
            else:
                if len(stack) > 0 and parenMap[stack[-1]] == c:
                    stack.pop()
                else:
                    return False

        return len(stack) == 0

File: 20.valid-parentheses.js
    const map = {
        "}": "{",
        "]": "[",
        ")": "(",
    };
/**
 * @param {string} s
 * @return {boolean}
 */
    var isValid = function(s) {
        const stack = [];
        for (const char of s) {
            if (Object.values(map).includes(char)) {
                stack.push(char);
            }
            else {
                if (stack.pop() !== map[char]) {
                    return false
                }
            }
        }
        return stack.length === 0;
    };
